packages/contract/src/DaimoAccount.sol

```diff
@@ -161,14 +181,36 @@ contract DaimoAccount is IAccount, UUPSUpgradeable, Initializable, IERC1271 {
     ) private view returns (bool) {
         if (signature.length < 1) return false;
 
-        // First bit identifies the keySlot
-        uint8 keySlot = uint8(signature[0]);
+        // First bit identifies the number of signatures
+        uint8 numSignatures = uint8(signature[0]);
+        if (numSignatures < signatureThreshold) return false;
+
+        // TODO: this requires all signatures be valid. What
+        // if 2 sigs are required, 3 get passed in, and 1 is
+        // invalid. Should op succeed?
+        uint256 offset = 1;
+        for (uint256 i = 0; i < numSignatures;) {
+            uint8 keySlot = uint8(signature[offset]);
+            // TODO: is this even needed? can signatures vary in length?
```

Comment:
i assume but don't know that different kinds of signatures might have different lengths. all passkey sigs have the same length i think and i think maybe same as EOAs (since technically it's the same curve with different params? or .. something?) idk we'll figure out these things later lol